### PR TITLE
Add MinAres solver to iterative wrappers

### PR DESCRIPTION
**Pull request by mertkipcak**
## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API
  
## Additional context

A very minor change to add the minares solver from `Krylov.jl` to the iterative wrappers. The paper can be found [here](https://arxiv.org/abs/2310.01757)

### CODE REVIEW

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
>   UndefVarError: `KrylovJL_MINARES` not defined

Export is missing.

File: test/basictests.jl [hunk 1]
```diff
@@ -274,7 +274,8 @@ end
             ("GMRES", KrylovJL_GMRES(kwargs...)),
             ("GMRES_prec", KrylovJL_GMRES(; precs, ldiv = false, kwargs...)),
             # ("BICGSTAB",KrylovJL_BICGSTAB(kwargs...)),
-            ("MINRES", KrylovJL_MINRES(kwargs...))
+            ("MINRES", KrylovJL_MINRES(kwargs...)),
+            ("MINARES", KrylovJL_MINARES(kwards...))
```
Comment (ChrisRackauckas) [inline test/basictests.jl:278]:
```suggestion
            ("MINARES", KrylovJL_MINARES(kwargs...))
```

**Comment by mertkipcak** [comment]
Hi Chris, thanks for fix on the typo and the export! Please let me know if I need to update anything else

**Comment by ChrisRackauckas** [comment]
No problem. Works on LTS, v1 tests are just an Enzyme thing, so this is good to go.

**Comment by mertkipcak** [comment]
@ChrisRackauckas, quick question. What is usually the release schedule? If it is not close by, I'll just use my fork for now

**Comment by ChrisRackauckas** [comment]
done

**Pull request merged**